### PR TITLE
feat(ssr): add ionic angular server

### PR DESCRIPTION
**Pull request by adamdbradley**

### CODE REVIEW

File: core/package.json [hunk 1]
```diff
@@ -99,7 +100,7 @@
   "bugs": {
     "url": "https://github.com/ionic-team/ionic/issues"
   },
-  "homepage": "https://github.com/ionic-team/ionic#readme",
+  "homepage": "https://ionicframework.com/",
```
Comment (manucorporat) [inline core/package.json]: https://ionicframework.com/docs

?

File: core/scripts/cdn-loader.js [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
-    const mod = doc.createElement('script');
-    mod.setAttribute('type', 'module');
-    mod['data-opts'] = opts;
-    mod.src = '__CDN_LOADER_URL__/dist/ionic/ionic.esm.js';
 
-    const legacy = doc.createElement('script');
-    legacy.setAttribute('nomodule', '');
-    legacy['data-opts'] = opts;
-    legacy.src = '__CDN_LOADER_URL__/dist/ionic/ionic.js';
+exports.applyPolyfills = function() { return Promise.resolve() };
```
Comment (manucorporat) [inline core/scripts/cdn-loader.js]: what's the difference?

File: core/scripts/copy-cdn-loader.js [hunk 3]
```diff
@@ -6,10 +6,12 @@
...
 
-const version = process.argv[2];
+let version = process.argv[2];
 
-if (!version) {
-  throw new Error('version arg missing');
+if (version) {
+  version = '@' + version;
+} else {
+  version = '';
```
Comment (manucorporat) [inline core/scripts/copy-cdn-loader.js]: why version-less?

File: angular/test/test-app/tslint.json [hunk 4]
```diff
@@ -1,36 +1,16 @@
 {
-  "rulesDirectory": [
-    "node_modules/codelyzer"
-  ],
+  "extends": "tslint:recommended",
+  "rulesDirectory": ["codelyzer"],
   "rules": {
-    "arrow-return-shorthand": true,
```
Comment (manucorporat) [inline angular/test/test-app/tslint.json]: why all this changes?

File: angular/test/test-app/src/main.ts [hunk 5]
```diff
@@ -8,5 +8,8 @@ if (environment.production) {
   enableProdMode();
 }
 
-platformBrowserDynamic().bootstrapModule(AppModule)
-  .catch(err => console.error(err));
+document.addEventListener('DOMContentLoaded', () => {
```
Comment (manucorporat) [inline angular/test/test-app/src/main.ts]: starter from ngCLI does not do that, why us?

File: angular/test/test-app/browserslist [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# This file is used by the build system to adjust CSS and JS output to support the specified browsers below.
```
Comment (manucorporat) [inline angular/test/test-app/browserslist]: this file is already in test-app/src

File: core/scripts/copy-cdn-loader.js (hunk 3 above)
Comment (adamdbradley) [reply core/scripts/copy-cdn-loader.js]: internal testing

File: angular/test/test-app/tslint.json (hunk 4 above)
Comment (adamdbradley) [reply angular/test/test-app/tslint.json]: ng8 updates

File: core/package.json (hunk 1 above)
Comment (adamdbradley) [reply core/package.json]: For the core? For the ionic/docs I have it at `https://ionicframework.com/docs`

File: angular/test/test-app/tslint.json (hunk 4 above)
Comment (mhartington) [reply angular/test/test-app/tslint.json]: as part of the ng8 update, codelyzer also released a 5.0 which included some rules being changed/renamed. This just followed the standard ng8 app starter.

File: angular/test/test-app/src/main.ts (hunk 5 above)
Comment (mhartington) [reply angular/test/test-app/src/main.ts]: This is for SSR. To make sure that the DOM is ready before bootstrapping the client side app.

File: angular/test/test-app/browserslist (hunk 6 above)
Comment (mhartington) [reply angular/test/test-app/browserslist]: This was moved from src to root.

**Comment by MTXDevp** [comment]
Is it currently working?

**Pull request merged**